src/dispatcher.rs

```diff
@@ -31,6 +57,75 @@ impl Handler<CollectionInfo> for DBExecutor {
     }
 }
 
+impl Handler<GetBso> for DBExecutor {
+    type Result = Result<Option<BSO>, Error>;
+
+    fn handle(&mut self, msg: GetBso, _: &mut Self::Context) -> Self::Result {
```

Comment:
Sorry I missed this earlier today, Phil (I just turned on slack notifications on this repo to #push-dev-bots so I get nagged there).

The most immediate comment is that we're using SyncExecutor (sync actors) for DBExecutor. So all this code can just be synchronous (and I wish I told you hours ago)!

So though the handlers deal with an asynchronous call (waiting for the db_executor to finish) under the covers the message to the DBExecutor goes to a thread safe mailbox, then gets picked up by one of a number of DBExecutors in a thread pool.

Pretty much along these lines: https://actix.rs/docs/databases/